ProteoformSuiteInternal/ProteoformCommunity.cs

```diff
@@ -80,10 +82,53 @@ public bool allowed_ee_relation(ExperimentalProteoform pf1, ExperimentalProteofo
             return pf1.modified_mass >= pf2.modified_mass
                 && pf1 != pf2
                 && (!Lollipop.neucode_labeled || pf1.lysine_count == pf2.lysine_count)
-                && pf1.modified_mass - pf2.modified_mass <= Lollipop.ee_max_mass_difference
-                && Math.Abs(pf1.agg_rt - pf2.agg_rt) <= Lollipop.ee_max_RetentionTime_difference;
-                //where ProteoformRelation.mass_difference_is_outside_no_mans_land(pf1.modified_mass - pf2.modified_mass)
-                //putative counts include no-mans land, currently
+                && allowed_mass_difference(pf1.modified_mass, pf2.modified_mass, ProteoformComparison.ee)
+                && matching_RT(pf1.all_RTs, pf2.all_RTs, Lollipop.ee_max_RetentionTime_difference);
+
+            //where ProteoformRelation.mass_difference_is_outside_no_mans_land(pf1.modified_mass - pf2.modified_mass)
+            //putative counts include no-mans land, currently
+        }
+
+        public bool allowed_ef_relation(ExperimentalProteoform pf1, ExperimentalProteoform pf2)
+        {
+            return pf1.modified_mass >= pf2.modified_mass
+            && pf1 != pf2
+            && (!Lollipop.neucode_labeled || pf1.lysine_count != pf2.lysine_count)
+            && (Lollipop.neucode_labeled || !matching_RT(pf1.all_RTs, pf2.all_RTs, Lollipop.ee_max_RetentionTime_difference * 2))
+            && allowed_mass_difference(pf1.modified_mass, pf2.modified_mass, ProteoformComparison.ef)
+            && (!Lollipop.neucode_labeled || matching_RT(pf1.all_RTs, pf2.all_RTs, Lollipop.ee_max_RetentionTime_difference));
+        }
+
+        public bool allowed_mass_difference(double pf1_mass, double pf2_mass, ProteoformComparison comparison)
+        {
+            if (comparison == ProteoformComparison.et || comparison == ProteoformComparison.ed)
+            {
+                if (Lollipop.notch_search_et)
+                {
+                    foreach (double mass in Lollipop.notch_masses_et)
```

Comment:
if you use `foreach (double mass in comparison == ProteoformComparison.et || comparison == ProteoformComparison.ed ? Lollipop.notch_masses_et : Lollipop.notch_masses_ee)`

This would simplify the code here.